### PR TITLE
feat: add minio tenant helm chart conditionally

### LINKED ISSUE #177: Charts[Network Node]: Ensure Mirror Uploader Uploads Stream Files To S3 Compatible Minio Bucket
User story
-----------
**As a** FST user
**I want**
various node stream files be uploaded into a S3 compatible bucket (compressed/uncompressed)
**So that**
I can run mirror node using those files later

Assumptions
---------------
- User is running a local kind cluster
- User is able to run `helm` and `kubectl` command
- User is able to deploy minio using helm chart

Requirements
---------------
- Setup buckets using Minio
- Configure mirror uploader to use minio buckets
- Ensure it is uploading files (compressed/uncompressed)
- Optional: Ensure those stream files are not corrupted (can be used to start mirror nodes)

Notes
---------
- Run network node (use the bash scripts that we are going to create to install and run node software)
- Ensure it is generating stream files
- Ensure stream files are being uploaded to the bucket (compressed/uncompressed)
- Optional: start the mirror node container using these stream files **(See the testing section in the comment below)**

### PR DESCRIPTION
**Pull request by deepak-swirlds**
## Description

This pull request changes the following:

- Add minio chart (conditionally, only if uploader sidecars are used)
- Add mirror node chart (conditionally, only if uploader sidecars are used)

### Related Issues

- Closes #177

### CODE REVIEW

File: charts/hedera-network/values.yaml [hunk 1]
```diff
@@ -160,10 +161,37 @@ defaults:
...
+      name: myminio-env-configuration
+    certificate:
+      requestAutoCert: false
+
 # hedera node configuration
 # Only the name of the node is required. The rest of the configuration will be inherited from `defaults` section
 hedera:
   nodes:
     - name: node0
+      id: 0.0.3
```
Comment (leninmehedy) [inline charts/hedera-network/values.yaml:193]: `id` should be renamed to `accountId` for clarity.

Comment (jeromy-cannon) [reply charts/hedera-network/values.yaml:190]: also, might add some comments on what the allowable values can be for this variable.

File: charts/hedera-network/values.yaml [hunk 2]
```diff
@@ -16,4 +17,27 @@ terminationGracePeriodSeconds: 10
 # Then the chart can be deployed using command below:
 # e.g. helm install fst ./charts/hedera-network --values ./charts/hedera-network/nodes.yaml
 hedera:
+  uploaderSidecars: true
   nodes: []
+
+# This configured the minio tenant subchart
+# Reference for configuration: https://github.com/minio/operator/blob/master/helm/tenant/values.yaml
+tenant:
```
[the remark this answers is not in the document]
Comment (deepak-swirlds) [reply charts/hedera-network/values.yaml:163]: I verified that this way of passing values does not work
I have to pass like below to make it available to the subchart
tenant: 
 ...

File: charts/hedera-network/values.yaml [hunk 3]
```diff
@@ -16,4 +17,27 @@ terminationGracePeriodSeconds: 10
...
+    #TODO: use helm random alphanumeric generator, or is it ok to hardcode
+    accessKey: minio
+    secretKey: minio123
+    buckets:
+    - name: bucket1
+      #objectLock: false        # optional
+  tenant:
+    name: myminio
+    pools:
+      - servers: 4
```
[the remark this answers is not in the document]
Comment (deepak-swirlds) [reply charts/hedera-network/values.yaml]: this is fixed now

File: charts/hedera-network/Chart.yaml [hunk 4]
```diff
@@ -22,3 +22,11 @@ version: 0.1.0
...
 # It is recommended to use it with quotes.
 appVersion: "0.40.2"
+
+# helm dependency update
+dependencies:
+  - name: tenant
+    version: 5.0.6
+    repository: https://operator.min.io/
+    # TODO:  is this a good enough name, if we want to group enable/disable the uploader sidecars as a group
+    condition: hedera.uploaderSidecars
```
[the remark this answers is not in the document]
Comment (deepak-swirlds) [reply charts/hedera-network/Chart.yaml]: using an alias now

File: charts/hedera-network/templates/sidecars/_account-balance-uploader.tpl [hunk 5]
```diff
@@ -14,10 +15,13 @@
     - /usr/local/bin/mirror.py
     - --linux
     - --watch-directory
-    - /opt/hgcapp/accountbalance
+    - /opt/hgcapp/accountBalances
+    - --s3-endpoint
+    - http://myminio-hl:9000
```
Comment (nathanklick) [inline charts/hedera-network/templates/sidecars/_account-balance-uploader.tpl:20]: This should not be hardcoded.

File: charts/hedera-network/templates/sidecars/_record-stream-uploader.tpl [hunk 6]
```diff
@@ -14,12 +15,15 @@
     - /usr/local/bin/mirror.py
     - --linux
     - --watch-directory
-    - /opt/hgcapp/recordstream
+    - /opt/hgcapp/recordStreams
     - --csv-stats-directory
-    - /opt/hgcapp/recordstream/uploader-stats
+    - /opt/hgcapp/recordStreams/uploader-stats
+    - --s3-endpoint
+    - http://myminio-hl:9000
```
Comment (nathanklick) [inline charts/hedera-network/templates/sidecars/_record-stream-uploader.tpl:22]: Should not be hardcoded.

File: charts/hedera-network/values.yaml [hunk 7]
```diff
@@ -162,10 +160,37 @@ defaults:
...
 
+# This configures the minio tenant subchart
+# Reference for configuration: https://github.com/minio/operator/blob/master/helm/tenant/values.yaml
+minio-server:
+  secrets:
+    name: myminio-env-configuration
+    # MinIO root user and password
+    #TODO: use helm random alphanumeric generator, or is it ok to hardcode ?
+    accessKey: minio
+    secretKey: minio123
```
Comment (nathanklick) [inline charts/hedera-network/values.yaml:171]: In response to the TODO comments on this line, we should use unique generated secrets stored in the kubernetes cluster and never expose them or hard code them in the template.

File: charts/hedera-network/templates/sidecars/_account-balance-uploader.tpl (hunk 5 above)
Comment (deepak-swirlds) [reply charts/hedera-network/templates/sidecars/_account-balance-uploader.tpl:20]: do you think I should externalize it in the values.yaml or did you have something else in mind ?

File: charts/hedera-network/values.yaml (hunk 7 above)
Comment (deepak-swirlds) [reply charts/hedera-network/values.yaml]: Fixed, a new secret minio-secrets has been introduced which has the secrets,
this secret is autogenerated by helm.
The secret values are also used in in side car uploaders to connect to minio

File: charts/hedera-network/values.yaml [hunk 8]
```diff
@@ -162,10 +160,34 @@ defaults:
...
+      requestAutoCert: false
+
 # hedera node configuration
 # Only the name of the node is required. The rest of the configuration will be inherited from `defaults` section
 hedera:
   nodes:
     - name: node0
+      accountId: 0.0.3
     - name: node1
+      accuntIdd: 0.0.4
```
Comment (nathanklick) [inline charts/hedera-network/values.yaml:191]:
```suggestion
      accuntId: 0.0.4
```

**Review by nathanklick — APPROVED** [review]

**Pull request merged**